tests/reqRep.cpp

```diff
@@ -0,0 +1,238 @@
+
+/* Copyright (c) 2017, Human Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#define BOOST_TEST_MODULE zeroeq_req_rep
+
+#include "common.h"
+
+#include <servus/servus.h>
+#include <servus/uri.h>
+
+#include <chrono>
+#include <thread>
+
+namespace
+{
+static const float TIMEOUT = 100.f; // milliseconds
+
+void runOnce(zeroeq::Server& server, const test::Echo& expected)
+{
+    bool handled = false;
+    server.handle(test::Echo::IDENTIFIER(),
+                  [&](const void* data, const size_t size) {
+                      BOOST_CHECK(data);
+                      BOOST_CHECK(!handled);
+
+                      const auto got = test::Echo::create(data, size);
+                      BOOST_CHECK_EQUAL(got, expected);
+
+                      handled = true;
+                      return test::Echo("Jumped over the lazy dog").toBinary();
+                  });
+
+    BOOST_CHECK(!handled);
+    server.receive(TIMEOUT);
+    BOOST_CHECK(handled);
+}
+
+bool running;
+};
+}
+
+BOOST_AUTO_TEST_CASE(serializable)
+{
+    test::Echo echo("The quick brown fox");
+
+    zeroeq::Server server(zeroeq::NULL_SESSION);
+    zeroeq::Client client(zeroeq::URI(server.getURI()));
+
+    std::thread thread([&] { runOnce(server, echo); });
+
+    bool handled = false;
+    client.get(echo.getTypeIdentifier(), [&](const void* data,
```

Comment:
This is something that was not clearly specified in the feature request. Client get/set methods were mentioned, but only in a example, and the example doesn't match this test. What's the value of this get method over Client::request, because I don't see many differences.